### PR TITLE
[RFC / Discussion] python3 compat mode

### PR DESCRIPTION
**Pull request by lethliel**
This is for discussing the python3 compatibility changes.

### CODE REVIEW

File: osc/build.py [hunk 1]
```diff
@@ -1143,7 +1143,10 @@ def __str__(self):
         if bi.installonly_list:
             rpmlist.append('installonly: ' + ' '.join(bi.installonly_list) + '\n')
 
-    rpmlist_file = NamedTemporaryFile(prefix='rpmlist.')
+    if sys.version_info >= (3, 0):
+      rpmlist_file = NamedTemporaryFile(mode='w+t', prefix='rpmlist.')
```
Comment (Conan-Kudo) [inline osc/build.py]: This variant works in both Python 2 and Python 3, so we can simplify to that.

File: osc/util/packagequery.py [hunk 2]
```diff
@@ -56,7 +56,10 @@ def read(self, all_tags = False, *extra_tags):
     @staticmethod
     def query(filename, all_tags=False, extra_rpmtags=(), extra_debtags=(), self_provides=True):
         f = open(filename, 'rb')
-        magic = f.read(7)
+        if sys.version_info >= (3, 0):
+            magic = f.read(7).decode('latin-1')
```
Comment (Conan-Kudo) [inline osc/util/packagequery.py]: Why are we forcing `latin-1` here instead of `utf-8`?

File: osc/build.py (hunk 1 above)
Comment (lethliel) [reply osc/build.py]: yes. you are right.

File: osc/util/packagequery.py (hunk 2 above)
Comment (lethliel) [reply osc/util/packagequery.py]: This will be handled completely different. I do not decode here at all. I will work with the bytes

**Comment by lethliel** [comment]
This pull request is obsolete. It will be handled in an own branch until everything works and will be merged in the end. 

https://github.com/openSUSE/osc/tree/python3

**Pull request closed without merging**